Readd get_x11_console_tty supporting all current variants (#3645)

okurz opened the pull request:
I moved the distribution initialization to make sure the variables evaluated
in init_consoles calling get_root_console_tty already regard the updated
defaults, for example the value for default desktop.

This does not consider a display manager different then the default for a
certain desktop as well as other potential impacts but the approach is proven
to work for now.

Validation runs:
* SLE15 gnome: http://lord.arch/tests/7577
* SLE15 icewm: http://lord.arch/tests/7564
* Leap 42.3 QAM icewm: http://lord.arch/tests/7562
* Tumbleweed gnome: http://lord.arch/tests/7563

okurz commented:
see #3645

coolo commented:
this also needs to be specific to DESKTOP=gnome. xdm does not have this problem^Hfeature

DimStar77 commented on lib/utils.pm:
```diff
@@ -1382,7 +1382,7 @@ sub get_root_console_tty {
 =cut
 sub get_x11_console_tty {
-    return (sle_version_at_least('15')) ? 2 : 7;
+    return ((is_sle && sle_version_at_least('15')) || (is_leap && leap_version_at_least('15'))) ? 2 : 7;
```
the condition is wrong imho

It should be desktop = gnome, notauologin=1 and sle_version_at_least 15

okurz replied:
yes, that makes sense.

okurz replied:
trying with

```perl
(check_var('DESKTOP', 'gnome') && get_var('NOAUTOLOGIN') && \
    !(is_sle && !sle_version_at_least('15')  ) && \
    !(is_leap && !leap_version_at_least('15'))) ? 2 : 7
```

coolo commented:
please split this statement. this is impossible to read

coolo commented:
(and not just as travis suggests :)

okurz commented:
ok, so we reverted the original PR to fix the openSUSE TW and maintenance test failures so we are not in a hurry anymore. I re-added the original changes in an adjusted manner in this PR now so please take a look. Updated description.

@Vogtinator, @chuchingkai, @coolo please take a look

Vogtinator commented:
Looks ok to me, but for the long term a different approach might be better.
The openSUSE Plasma Wayland test installs a default Plasma on X session with autologin enabled and switches to wayland after login, which means the graphical session ends up on VT 3 (root-console on tty1 and user-console on tty2), but only after some modules are done running. So IMO there should be an official API (not the current hack in start_wayland_plasma5.pm) to allow setting the number of the graphical VT during the test run. Then the VT determination code can be moved into a separate file/module which makes it easier to understand and adapt.

coolo commented:
supporting a set_tty in the tty console sounds indeed necessary to cover all cases. Then we can set it in first_boot. But it would be boring if we ran out of options :)

coolo commented:
https://progress.opensuse.org/issues/25702

thehejik commented on lib/utils.pm:
```diff
@@ -1375,6 +1376,16 @@ sub get_root_console_tty {
+=cut
+sub get_x11_console_tty {
+    my $new_gdm = !(is_sle && !sle_version_at_least('15')) && !(is_leap && !leap_version_at_least('15'));
+    return (check_var('DESKTOP', 'gnome') && get_var('NOAUTOLOGIN') && $new_gdm) ? 2 : 7;
```
This broke CaaSP-controller test which is basically SLE12_SP2 support_server but with DISTRI=caasp, DESKTOP=gnome, NOAUTOLOGIN=1 set ---> $new_gdm = True

 it returns **2** but it should be **7**

thehejik replied:
should be fixed by https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/3677
